### PR TITLE
fix for #2667: put email-update-confirmation code into separate controller

### LINKED ISSUE #2667: ProfileController is not available if registration.confirmation is not enabled
If registration confirmation is not enabled, then `FOS\UserBundle\Controller\ProfileController` is not available, because `Resources/config/email_confirmation.xml` was not loaded and `fos_user.email_update_confirmation` service was not created.

> The service "fos_user.profiler.controller" has a dependency on a non-existent service "fos_user.email_update_confirmation".

I could temporary fix this error with configuration:

```yaml
fos_user:
    registration:
        confirmation:
            enabled: true
```

May be `confirmEmailUpdateAction` method should be moved out from `FOS\UserBundle\Controller\ProfileController` to a new controller?

### PR DESCRIPTION
**Pull request by azine**
Untagled some configuration dependencies between 
email_confirmation (upon registration) and 
confirm_email_update (when the users changes her email) to fix  #2667

### CODE REVIEW

**Comment by azine** [comment]
I moved the `confirmEmailUpdateAction` into a separate/new controller, re-grouped the configuration related to the email-update-confirmation-feature, fixed a typo and added a test for the routing.

DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! 
I didn't yet run this code in a fully installed application. I only just started updating my own application from Symfony 2.7 to 2.8 (and then later 3.x). I'll give this code a try once I have updated my app.
DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! DISCLAIMER!!! 

I you (@covex-nn or @XWB or @stof or anyone else) want to give it a try before that, you are very welcome to do so and give feedback. 

Cheers,
Dominik

File: Resources/config/profile.xml [hunk 1]
```diff
@@ -18,12 +18,10 @@
             <tag name="form.type" alias="fos_user_profile" />
         </service>
 
-        <service id="fos_user.profiler.controller" class="FOS\UserBundle\Controller\ProfileController" public="true">
```
Comment (azine) [inline Resources/config/profile.xml]: fixed typo

**Comment by covex-nn** [comment]
@azine without enabling `profile.email_update_confirmation` feature, `/profile` and `/profile/edit` work as before. And `/profile/confirm-email-update/qwerty` is not accessable with this error message:

> Controller not found: service "fos_user.confirm.email.update.controller" does not exist.

After enabling email update confirmation with only `profile.email_update_confirmation.enabled: true`, and after updating email in `/profile/edit` there was a error in `FOS\UserBundle\Mailer\Mailer::sendUpdateEmailConfirmation`

> Notice: Undefined index: template

This error occurred because, there is no `template` key in `$this->parameters`. But, there is `email_updating.template` key, and `$this->parameters['email_updating.template'] === '@FOSUser/Profile/email_update_confirmation.txt.twig'`

**Comment by covex-nn** [comment]
@azine i checked: second part was fixed =)

After updating email, `fos_user.email_canonical` was changed to new value, `fos_user.email` had old value.
Then after visiting `$confirmationUrl`, `fos_user.email` was updated to new value too, `fos_user.confirmation_token` was updated back to null.

So, it works!

**Comment by azine** [comment]
@covex-nn Thanks again for testing. Leaving the canonical value of the email with the new value while waiting for confirmation, was not intentional. => I fixed this unintended behaviour.

**Comment by covex-nn** [comment]
@azine I thought, that it was a feature. I wrote about it just because is was strange =)

With last commit, `fos_user.email` and `fos_user.email_canonical` are changed only after visiting `$confirmationUrl`. So, it works2! Hooray! =)

**Comment by azine** [comment]
@covex-nn cool! Thanks for the "testing service". :-)

IMO thiis PR would be ready to squash & merge. @XWB  or @stof : any objecttions? Anything I need to change?

**Comment by XWB** [comment]
Thank you guys.

**Pull request merged**